src/ShopPaymentTransactionBundle/objects/TPkgShopPaymentTransactionManagerEndPoint.class.php

```diff
@@ -619,10 +678,25 @@ public function getTransactionDataFromOrder(
             $dOrderTotal = $dOrderTotal + ($oItem->getAmount() * $oItem->getValue());
         }
 
-        $dVoucher = (-1 * $this->order->fieldValueVouchers) - ($this->getTransactionPositionTotalForType(
```

Comment:
This method is ridiculously long (170 lines!) - could you move the new lines to a separate method?

Reply:
done